dashboard/src/actions/authActions.js

```diff
@@ -12,7 +11,8 @@ export const makeLoginRequest = (details, navigate) => async (dispatch, getState
       type: TYPES.USER_NOTION_ALERTS,
       payload: [],
     });
-    const response = await API.post(API_ROUTES.LOGIN, {
+    const endpoints = {...getState().apiEndpoint.endpoints};
+    const response = await API.post(endpoints?.api?.login, {
```

Comment:
I'm not sure about the `?.` usage here. What do we *want* to happen if we somehow get here without having loaded the endpoints? I suspect that should be a failure, rather than trying the `API.post` with `undefined` as a URL.

I understand that this is a "safe" general pattern for navigating an object, and doesn't hurt if all the expected references are there; but this makes me think about what happens if we can't load. The App component is throwing up a TOAST message; but maybe we should try to go further and completely lock up the UI with the warning message until they refresh (and it loads successfully)?

Reply:
If `fetch endpoints` request is successful, _login_ or _register_ buttons would be enabled.